invenio_rdm_records/fixtures/vocabularies.py

```diff
@@ -293,64 +286,135 @@ def get_records_by_vocabulary(self, vocabulary_id):
         for id_, entry in self.read():
             if vocabulary_id != id_:
                 continue
-            for record in self.iter_datafile(entry["data-file"]):
+            for record in entry:
                 yield record
 
     def load(self, ignore=None):
-        """Load the fixture.
+        """Load the whole fixture.
 
         ignore: iterable of ids to ignore
 
         For subjects (or any vocabulary with a dict of data-file), the id
         that counts is ``<vocabulary id>.<dict key>``.
+
+        Returns all vocabulary ids loaded so far.
         """
         ids = set().union(ignore) if ignore else set()
 
         for id_, entry in self.read():
-            # TODO: We need to carefully think about what the workflow for
-            #       adding new subjects later is and then modify the code to
-            #       allow for it.
-            if id_ not in ids:
-                try:
-                    self.create_vocabulary_type(id_, entry)
-                except IntegrityError:
-                    # Can only get here when running the code multiple times
-                    # on same database. To keep the code simple for now, we
-                    # simply skip previously existing types.
-                    current_app.logger.info(
-                        f"Skipping creation and loading of pre-existing {id_}"
-                    )
-                    continue
-
-            if isinstance(entry["data-file"], dict):
-                # The subvocabularies may not have been loaded
-                for key, filepath in entry["data-file"].items():
-                    sub_id = f"{id_}.{key}"
-                    if sub_id not in ids:
-                        self.load_datafile(id_, filepath)
-                        ids.add(sub_id)
-                ids.add(id_)
-            elif id_ not in ids:
-                self.load_datafile(id_, entry["data-file"])
-                ids.add(id_)
+            ids.update(
+                entry.load(self._identity, ignore=ids, delay=self._delay)
+            )
 
         return ids
 
-    def create_vocabulary_type(self, id_, entry):
+
+class GenericVocabularyEntry:
```

Comment:
This and the class below seem to share a common structure, is it worth it to create an interface and maybe move some functionality there? Might be useful when doing this for affiliations.

Reply:
3-strike refactor. The loading of affiliations will tell us once we get to it.